### PR TITLE
which: don't fail on malformed paths in PATH

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

**Comment by bfontaine** [comment]
I added a failing test which shows the issue: it fails without this PR, succeed with it.

**Comment by bfontaine** [comment]
I wonder if we should warn users in `brew doctor` for this.

File: Library/Homebrew/test/test_utils.rb [hunk 1]
```diff
@@ -6,6 +7,30 @@ def test_put_columns_empty
     assert_silent { puts_columns [] }
   end
 
+  def test_which
+    cmd = Tempfile.new "foo"
```
Comment (xu-cheng) [inline Library/Homebrew/test/test_utils.rb]: You don't need to use `Tempfile` in the test. Instead you can do:
```
def test_which
  dir = Pathname.new(mktmpdir)
  cmd = dir/"foo"
  cmd.chmod 0744
  assert_equal cmd, which("foo", dir)
ensure
  dir.rmtree
end
```

Comment (bfontaine) [inline Library/Homebrew/test/test_utils.rb]: Thanks @xu-cheng, fixed. Any remark outside of that?

Comment (xu-cheng) [inline Library/Homebrew/test/test_utils.rb]: No. You are all good :+1:

**Pull request closed without merging**